File: convert-integer-to-the-sum-of-two-no-zero-integers/convert-integer-to-the-sum-of-two-no-zero-integers.cpp
class Solution {
public:
  vector<int> getNoZeroIntegers(int n) {
    auto valid = [](int x) {
      if (!x) return false;
      while (x) {
        if (x % 10 == 0) return false;
        x /= 10;
      }
      return true;
    };
    
    for (int i = 1; i <= n / 2; ++i)
      if (valid(i) && valid(n - i)) 
        return {i, n - i};
    return {};
  }
};

File: 1440-convert-integer-to-the-sum-of-two-no-zero-integers/convert-integer-to-the-sum-of-two-no-zero-integers.py
class Solution:
    def getNoZeroIntegers(self, n: int) -> List[int]:
        li = []
        for i in range(1, n//2 +1):
            a = str(i)
            b = str(n - i)
            if '0' not in a and '0' not in b:
                li.append(int(a))
                li.append(int(b))
                return li
        
        